index.js

```diff
@@ -81,7 +83,14 @@ module.exports = function (opts) {
 					deleteRequireCache(modId);
 				});
 			}
-			jasmine.addReporter(new SilentReporter(cb, errorOnFail));
+			jasmine.addReporter(new SilentReporter(function (error) {
+				if (error) {
+					cb(error);
+				} else {
+					self.emit('end');
+					cb();
```

Comment:
Not sure if this is possible with jasmine, but I think a better pattern might be to have the Transform stream process all the test files and attach metadata, then create a Writable stream that is the reporter so you can do something like:
```js
gulp.src('**/*.js').pipe(jasmine()).pipe(jasmine.report())
```